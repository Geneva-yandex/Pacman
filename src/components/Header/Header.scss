@import '../../styles/variables';

.Header {
  &__nav-list {
    height: 150px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__nav-item {
    a {
      color: $grey-light;
      font-size: 20px;
      font-weight: 700;
      text-transform: uppercase;
      transition: opacity $short-effect;
      text-decoration: none;

      &:hover {
        text-decoration: none;
        opacity: 0.64;
      }
    }

    &--root a {
      color: $yellow;
      font-size: 30px;
    }
  }
}
