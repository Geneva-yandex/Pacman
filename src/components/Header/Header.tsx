import * as React from 'react';
import bem from 'easy-bem';
import {Link} from 'react-router-dom';
import routes from '../../pages/index';
import {RouteType} from '../Router/types';
import './Header.scss';

const b = bem('Header');
export default class Header extends React.PureComponent {
    render() {
        return (
            <header className={b()}>
                <nav className={'container-fluid'}>
                    <ul className={b('nav-list')}>
                        {routes
                            .filter(route => route.isNavVisible)
                            .map(route => this.renderNavItem(route))
                        }
                    </ul>
                </nav>
            </header>
        )
    }

    renderNavItem(route: RouteType) {
        return (
            <li
                key={route.id}
                className={b('nav-item', {
                    [route.id]: true
                })}
            >
                <Link to={route.path}>
                    {route.title}
                </Link>
            </li>
        )
    }
}