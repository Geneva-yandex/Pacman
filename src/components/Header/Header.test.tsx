import * as React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';

describe('Header', () => {
    test('Header component renders correctly', () => {
        const renderedComponent = shallow(<Header />);
        expect(renderedComponent).toMatchSnapshot();
    });
});

