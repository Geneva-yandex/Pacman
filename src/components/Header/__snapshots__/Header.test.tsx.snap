// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component renders correctly 1`] = `ShallowWrapper {}`;
